backend/app/vending/__init__.py

```diff
@@ -568,6 +581,57 @@ def redeem_token(
     return RedemptionResult(status="failure", reason="invalid")
 
 
+# Tax and other real-world problems are associated with things like an
+# application's type, licence, etc.
+# This heuristic tries to tell us about the app, and why we made that decision
+@router.get("app/{appid}/info")
+def app_info(appid: str) -> VendingApplicationInformation:
+    """
+    This determines the vending info for the app and returns it
+    """
+
+    appstream = get_json_key(f"apps:{appid}")
+    if appstream is None:
+        raise HTTPException(status_code=404, detail=f"Application {appid} not found")
+    print(f"Found {appid}: {repr(appstream)}")
+
+    kind = "GENERIC"
+    kind_reason = "unknown"
+    foss = False
+    foss_reason = "unknown"
+
+    foss_licences = ["gpl-", "lgpl-", "bsd", "mit", "apache"]
```

Comment:
So it turns out that branding apps as PROPRIETARY is somewhat controversial (see https://gitlab.gnome.org/GNOME/gnome-software/-/issues/2004 and https://mastodon.social/@richardfontana/109492524538719614 ) and so we might want to be a little more cautious about how we approach this!

I like the idea of Flathub staying very consistent with GNOME Software and KDE Discover who both rely on https://github.com/ximion/appstream/blob/1522cafbb0cc6750c1664cae90bc8a0b6ab1dd24/src/as-spdx.c#L741 - is there a way we can either use the same license & exception lists or even use literally use libappstream?

Reply:
I've switched to using AppStream via GI - hopefully that'll be OK.